### PR TITLE
[jax2tf] Documentation for the native serialization mode

### PR DESCRIPTION
**Pull request by gnecula**

### CODE REVIEW

**Comment by gnecula** [comment]
@rxsang @maxwillzq PTAL

**Comment by mjsML** [comment]
@nouiz @pjannaty for viz

File: jax/experimental/jax2tf/README.md [hunk 1]
```diff
@@ -3,57 +3,72 @@
...
+the interoperation is by way of a StableHLO/HLO intermediate form. That is,
+the target function is lowered to HLO using standard native JAX or TensorFlow APIs,
+and then the HLO is invoked from TensorFlow or JAX. This has the advantage that
+it stays faithful to the semantics of the native target function, even as JAX and
+TensorFlow evolves. We refer to this usage mode as "native lowering" or
+"native serialization".
+
+Note that at the moment when using JAX native serialization the whole
+JAX compilation unit is wrapped with a single thin TensorFlow op, called `XlaCallModule`,
+that carries a serialized version of the HLO obtained from JAX. This
```
Comment (maxwillzq) [inline jax/experimental/jax2tf/README.md:30]: I have some confusion about the interchangeable use between "StableHLO" and "HLO" here.  I know we can legalize HLO to stableHLO.  For end users, can we only mention  StableHLO and keep StableHLO/HLO conversion as low level details ?

Comment (gnecula) [reply jax/experimental/jax2tf/README.md:30]: JAX uses now StableHLO, but TF uses HLO. That is why I decided to just use HLO when I talk about it being the interoperability in both directions. I will see if I can clarify.

File: jax/experimental/jax2tf/README.md [hunk 2]
```diff
@@ -3,57 +3,72 @@
...
+    to deserialize, compile, and execute the JAX program.
+  * the semantics of JAX program is preserved faithfully because it is entirely
+    captured by the HLO serialization.
+
+For backwards compatibility purposes, these interoperation APIs can be used also
+optionally without going through HLO.
+
+  * For calling JAX functions from TensorFlow,
+    it is possible to request that the JAX function be lowered not to HLO but to
+    TensorFlow ops. This can be achieved by setting `experimental_native_serialization=False`.
```
Comment (maxwillzq) [inline jax/experimental/jax2tf/README.md:48]: Great !  It is my first time know this.
By the way,  is it possible this `experimental_native_serialization=False` jax2tf.convert can replace current non_native_lowering approach ?  This question is not related this CL, just for curious.

File: jax/experimental/jax2tf/README.md [hunk 3]
```diff
@@ -3,57 +3,72 @@
...
+and then the HLO is invoked from TensorFlow or JAX. This has the advantage that
+it stays faithful to the semantics of the native target function, even as JAX and
+TensorFlow evolves. We refer to this usage mode as "native lowering" or
+"native serialization".
+
+Note that at the moment when using JAX native serialization the whole
+JAX compilation unit is wrapped with a single thin TensorFlow op, called `XlaCallModule`,
+that carries a serialized version of the HLO obtained from JAX. This
+op simply compiles the HLO and invokes it.
+While this may feel as not purely native, it has several advantages:
```
Comment (rxsang) [inline jax/experimental/jax2tf/README.md:32]: Instead of saying "this may feel as not purely native", we can just say "The reason we still wrap the lowered HLO in a TensorFlow op is..."

**Review by maxwillzq — APPROVED** [review]

File: jax/experimental/jax2tf/README.md [hunk 4]
```diff
@@ -986,19 +984,92 @@ streams it generates from different keys is less well understood.
...
+
+Applies to native serialization only.
+
+JAX native serialization uses the `XlaCallModule` TensorFlow op to host
+the StableHLO program obtained from JAX. This is a relatively
+new TensorFlow op and may not be supported by some tools. In fact,
+certain tools that need to do `tf.Graph` inspection and transformation
+cannot work when the whole JAX program is a single TensorFlow op.
+
+This is the case, for example, for the TFLite and TFjs converters.
```
Comment (rxsang) [inline jax/experimental/jax2tf/README.md:1025]: Should we say this is WIP?

**Review by rxsang — APPROVED** [review]
Thank you George!

File: jax/experimental/jax2tf/README.md (hunk 1 above)
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:30]: I have replaced every occurrence with StableHLO

File: jax/experimental/jax2tf/README.md [hunk 5]
```diff
@@ -3,57 +3,72 @@
...
+These APIs can be used in different modes, but in the default preferred mode
+the interoperation is by way of a StableHLO/HLO intermediate form. That is,
+the target function is lowered to HLO using standard native JAX or TensorFlow APIs,
+and then the HLO is invoked from TensorFlow or JAX. This has the advantage that
+it stays faithful to the semantics of the native target function, even as JAX and
+TensorFlow evolves. We refer to this usage mode as "native lowering" or
+"native serialization".
+
+Note that at the moment when using JAX native serialization the whole
+JAX compilation unit is wrapped with a single thin TensorFlow op, called `XlaCallModule`,
```
[the remark this answers is not in the document]
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:29]: Done

File: jax/experimental/jax2tf/README.md [hunk 6]
```diff
@@ -3,57 +3,72 @@
...
+  * For calling JAX functions from TensorFlow,
+    it is possible to request that the JAX function be lowered not to HLO but to
+    TensorFlow ops. This can be achieved by setting `experimental_native_serialization=False`.
+    Even in this case, most of JAX's internals are used as in the
+    native mode, except that each JAX primitive is lowered to a TensorFlow op instead of
+    an HLO op. The result is a function that is pretty much 1:1 with the HLO program
+    that would be obtained through native lowering. As a result, we can execute
+    the lowered code in TensorFlow eager mode for debugging,
+    or trace to a `tf.Graph` for consumption by tooling that understands TensorFlow
+    ops but does not yet work with HLO, e.g., TFLite and TFjs.
```
[the remark this answers is not in the document]
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:55]: I used TensorFlow.js, which I see is the official name of the project

File: jax/experimental/jax2tf/README.md (hunk 4 above)
Comment (gnecula) [reply jax/experimental/jax2tf/README.md:1025]: Done
